src/libs/actions/IOU.js

```diff
@@ -1668,6 +1674,59 @@ function sendMoneyViaPaypal(report, amount, currency, comment, managerID, recipi
     asyncOpenURL(Promise.resolve(), buildPayPalPaymentUrl(amount, recipient.payPalMeAddress, currency));
 }
 
+function approveMoneyRequest(expenseReport) {
+    const optimisticIOUReportAction = ReportUtils.buildOptimisticApprovedReportAction(expenseReport.total, expenseReport.currency, expenseReport.reportID);
+
+    const optimisticReportActionsData = {
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${expenseReport.reportID}`,
+        value: {
+            [optimisticIOUReportAction.reportActionID]: {
+                ...optimisticIOUReportAction,
+                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+            },
+        },
+    };
+    const optimisticIOUReportData = {
+        onyxMethod: Onyx.METHOD.MERGE,
+        key: `${ONYXKEYS.COLLECTION.REPORT}${expenseReport.reportID}`,
+        value: {
+            ...expenseReport,
+            lastMessageText: optimisticIOUReportAction.message[0].text,
+            lastMessageHtml: optimisticIOUReportAction.message[0].html,
+            stateNum: CONST.REPORT.STATE_NUM.SUBMITTED,
+            statusNum: CONST.REPORT.STATUS.APPROVED,
```

Comment:
Do we also need to update the chatReport, more specifically `iouReportID` and `hasOutstandingIOU`? If we don't and employees try to request money, we'd use the old reportID which is now approved and the request would likely fail, no?

Reply:
This action is being done from the principal's account which (only in TeachersUnite) doesn't have access to the policy expense chat, but only the expense report. 

>If we don't and employees try to request money, we'd use the old reportID which is now approved and the request would likely fail, no?

If the report is approved, we create a new report, see [here](https://github.com/Expensify/App/pull/24639/files#diff-17f969ae17c2a0f476494263758dbe0a535432f496b805345925e648d605915cR789).